pages/base.py

```diff
@@ -52,40 +57,95 @@ def __getattr__(self, element):
         else:
             raise ValueError('Cannot find element {}'.format(element))
 
+
 class BasePage(BaseElement):
     url = None
 
     def goto(self):
         self.driver.get(self.url)
 
-    def verify_page(self):
-        raise NotImplementedError
-
     def reload(self):
         self.driver.refresh()
 
-class OSFBasePage(BasePage):
-    url = settings.OSF_HOME
+
+class Navbar(BaseElement):
+
+    locators = {
+        'home_link': (By.XPATH, '//nav[@id="navbarScope"]/div/div[@class="navbar-header"]/div[@class="dropdown"]/ul[@role="menu"]/li/a[@href="' + settings.OSF_HOME + '"]'),
+        'preprint_link': (By.XPATH, '//nav[@id="navbarScope"]/div/div[@class="navbar-header"]/div[@class="dropdown"]/ul[@role="menu"]/li/a[@href="' + settings.OSF_HOME + '/preprints/"]'),
+        'registries_link': (By.XPATH, '//nav[@id="navbarScope"]/div/div[@class="navbar-header"]/div[@class="dropdown"]/ul[@role="menu"]/li/a[@href="' + settings.OSF_HOME + '/registries/"]'),
+        'meetings_link': (By.XPATH, '//nav[@id="navbarScope"]/div/div[@class="navbar-header"]/div[@class="dropdown"]/ul[@role="menu"]/li/a[@href="' + settings.OSF_HOME + '/meetings/"]'),
+        'search_link': (By.XPATH, '//div[@id="secondary-navigation"]/ul/li/a[@href="' + settings.OSF_HOME + '/search/"]'),
+        'support_link': (By.XPATH, '//div[@id="secondary-navigation"]/ul/li/a[@href="/support/"]'),
+        'donate_link': (By.XPATH, '//div[@id="secondary-navigation"]/ul/li/a[@href="https://cos.io/donate"]'),
+        'user_dropdown': (By.CSS_SELECTOR, '#secondary-navigation > ul > li:nth-child(5) > button'),
+        'user_dropdown_profile': (By.XPATH, '//div[@id="secondary-navigation"]/ul/li[@class="dropdown"]/ul/li/a[@href="/logout/"]'),
+        'user_dropdown_support': (By.XPATH, '//div[@id="secondary-navigation"]/ul/li[@class="dropdown"]/ul/li/a[@href="' + settings.OSF_HOME + '/support/"]'),
+        'user_dropdown_settings': (By.XPATH, '//div[@id="secondary-navigation"]/ul/li[@class="dropdown"]/ul/li/a[@href="/settings/"]'),
+        'sign_up_button': (By.LINK_TEXT, 'Sign Up'),
+        'sign_in_button': (By.LINK_TEXT, 'Sign In'),
+        'logout_link': (By.CSS_SELECTOR, '#secondary-navigation > ul > li.dropdown.open > ul > li:nth-child(4) > a'),
+        'current_service': (By.CSS_SELECTOR, '#navbarScope > div > div > div.service-home > a > span.current-service > strong')
+    }
 
     def __init__(self, driver):
-        super(OSFBasePage, self).__init__(driver)
-        self.navbar = self.Navbar(driver)
+        super(Navbar, self).__init__(driver)
+
+    def verify(self):
+        return len(self.find_elements(By.XPATH, '//nav[@id="navbarScope"]')) == 1
```

Comment:
Not sure this really matters, but may want to use `By.ID` here. Theoretically faster.